examples/js-datagrid/src/app/main-view/services/mock-data.service.ts

```diff
@@ -13,61 +12,59 @@ export class MockDataService{
   public subject: Subject<any> = new Subject<any>();
   public marketData: any;
   private market: Market;
-  private connected: Boolean;
+  private currentChannel: Channel | null;
+  private connected: Boolean = false;
   private connecting: Promise<void>;
 
-  constructor(@Inject(MESSAGE_ROUTER)private client: MessageRouter){
+  constructor(){
     this.market = new Market();
+    this.connecting = new Promise(async(resolve, reject) => {
+      try{
+        resolve(await this.checkFdc3Connection());
+      } catch(err) {
+        reject(err);
+      }
+    });
 
     interval(1000).subscribe(() => {
       this.marketData = this.market.generateNewMarketNumbers();
       this.subject.next(this.marketData);
     });
-
-    this.connecting = new Promise(async (resolve, reject) => {
-      try{
-        await this.checkMessageRouterConnection();
-        resolve(undefined);
-      }catch(exception){
-        console.log(exception);
-        reject(exception);
-      }
-    });
   }
 
-  public async publishSymbolData(symbol: any|undefined) {
-    await this.connecting;
-
-    if(symbol != undefined){
-      //Publishing to DataService
-      // let serialized = JSON.stringify(symbol);
-      // await this.client.publish('proto_select_marketData', serialized);
-
-      //Here we are connecting to the chart example without the DataService Example
-      let marketSymbol = this.market.createMarketSymbol(symbol);
-      let serialized = JSON.stringify(marketSymbol);
-      await this.client.publish('proto_select_monthlySales', serialized);
+  private async checkFdc3Connection(): Promise<void> {
+    if(!this.connected) {
+      this.currentChannel = await window.fdc3.getCurrentChannel();
+      if (!this.currentChannel) {
+        await window.fdc3.joinUserChannel("default");
+      }
+      this.connected = true;
     }
   }
 
-  public async requestSymbolData(topic: string){
-    await this.connecting;
-
-    await this.client.subscribe(topic, (message: TopicMessage) => {
-
-      const payload = JSON.parse(message.payload!);
+  public async publishSymbolData(symbol: any|undefined): Promise<void> {
+    if(symbol){
+      await this.connecting;
+      //TODO: involve DataService to generate the data for the chart
+      let marketSymbol = this.market.createMarketSymbol(symbol);
 
-      //TODO: Get live data from DataService example.
-      const symbol = payload.symbol;
-      const buyData = payload.buy;
-      const sellData = payload.sell;
-    });
-  }
+      const context: Context = {
+        type: 'fdc3.instrument',
+        id: {
+          ticker: marketSymbol?.symbol,
+          buyData: marketSymbol?.buy,
+          sellData: marketSymbol?.sell
+        }
+      }
 
-  private async checkMessageRouterConnection(){
-    if(!this.connected){
-      await this.client.connect();
-      this.connected = true;
+      await window.fdc3.broadcast(context);
     }
   }
+
+  // public async requestSymbolData(topic: string) : Promise<void>{
```

Comment:
Don't leave commented code

Reply:
that was on purpose as per It should (in a later state) involve DataService.
I'll remove it then!